src/engine/symbolic_state.rs

```diff
@@ -13,6 +13,14 @@ use petgraph::{
 use riscu::Instruction;
 use std::{collections::HashMap, fmt, ops::Index};
 
+#[derive(Debug, Clone, Copy)]
+pub enum Query {
```

Comment:
I am probably just blind, but I cannot find where this new enum is used, could you point me to it? Otherwise lets remove it, if it is not used.

Reply:
This is indeed legacy code. I'll remove it